aws/resource_aws_cognito_identity_provider.go

```diff
@@ -0,0 +1,181 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoIdentityProviderCreate,
+		Read:   resourceAwsCognitoIdentityProviderRead,
+		Update: resourceAwsCognitoIdentityProviderUpdate,
+		Delete: resourceAwsCognitoIdentityProviderDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"attribute_mapping": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"idp_identifiers": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"provider_details": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
```

Comment:
Not sure if this was intentional, but it does not seem like we should be ignoring this from the API if its not in the Terraform configuration. I'll double check from the acceptance testing. Also, the API/SDK documentation seems to imply this should be `Required: true`.